File: Programacion1/ej_programacion/ej_string/ej7.py
'''
Diseñar una función que reciba como parámetro tres cadenas, la primera será una frase y deberá buscar si existe 
la palabra que recibe como segundo parámetro y reemplazarla por la tercera.
Created on 27 Nov 2020

@author: estudiante
'''
def reemplazarPalabra(texto, palabra, reemplazo):
     textoNuevo=""
     contador=0
     for i in texto:
         print(i)
         if texto[i] == palabra[0]:
             for c in range(0, len(palabra)):
                 
             print("hola")
         print(texto[i])
reemplazarPalabra("hola como feo esta chaval, una pena ser tan feo", "feo", "malito")

File: programacionJava/programacionJava/src/ejerciciosString1/ej7.java
package ejerciciosString1;

import java.util.ArrayList;
import java.util.List;

public class ej7 {

	public static String prueba() {
		return "HOla";
	}
	
	
	public static void main(String[] args) {
		/*
		 * Diseñar una función que reciba como parámetro tres cadenas, la primera será una frase y 
		 * deberá buscar si existe la palabra que recibe como segundo parámetro y reemplazarla por la 
		 * tercera
		 */
		System.out.println(cambiaCadena("hola que haces", "haces", "tal"));
		
		
	}
	public static String cambiaCadena(String texto, String palabra, String reemplazo)
	{
		 List<String> lista = new ArrayList<String>();
		 String palabraCambiante="";
		 String textoNuevo="";
		 
		for (int i =0; i<texto.length();i++)
		{
			if (texto.charAt(i)!=' ')
			{
				palabraCambiante=palabraCambiante+texto.charAt(i);
			}
			else {
				lista.add(palabraCambiante);
				lista.add(" ");
				palabraCambiante="";
			}
		}
		for (int i=0;i<lista.size();i++) {
			if (lista.get(i).equals(palabra)) {
				lista.set(i, reemplazo);
			}
		}
		for (int i=0;i<lista.size();i++) {
			textoNuevo=textoNuevo+lista.get(i).toString();
		}
		return textoNuevo;
	}
	
}
